### PR TITLE
Fix bug where 'x = a if cond else b' was incorrectly parsed

### PR DESCRIPTION
**Pull request by pmbethe09**
Added a test to confirm.

### CODE REVIEW

File: core/testdata/045.golden [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
```
Comment (niltonvolpato) [inline core/testdata/045.golden]: Maybe you should also add a test case checking that "if a = b: ..." remains the same

Comment (pmbethe09) [inline core/testdata/045.golden]: Laurent reminded me that 
"if a = b: ..."
is not valid in Python

Comment (niltonvolpato) [inline core/testdata/045.golden]: Oh, true :)

**Comment by laurentlb** [comment]
LGTM

**Pull request merged**